chia/data_layer/data_layer.py

```diff
@@ -167,23 +169,25 @@ async def get_root_history(self, store_id: bytes32) -> List[SingletonRecord]:
         for record in records:
             if prev is None or record.root != prev.root:
                 root_history.append(record)
+                prev = record
         return root_history
 
     async def _validate_batch(
         self,
         tree_id: bytes32,
         to_check: List[SingletonRecord],
+        last_checked_hash: bytes32,
```

Comment:
i feel like the caller should filter this.

Reply:
refactored `_validate_batch` to include the first check as well, so we don't have to pass this in.